src/index.ts

```diff
@@ -200,6 +211,66 @@ class SvgChunkWebpackPlugin {
 		}
 	}
 
+	/**
+	 * injects the sprite map related to the entry in the related html generated by html-webpack-plugin
+	 * @param {} compilation
+	 * @param {Array<Sprite>} sprites
+	 * @returns {Promise<void>}
+	 */
+	async handleHtmlWebpackPlugin(compilation:Compilation, sprites:Array<Sprite>): Promise<void> {
+		let HtmlWebpackPluginImport;
+		try {
+			// @ts-ignore
+			HtmlWebpackPluginImport = await import('html-webpack-plugin');
+		} catch (e) {
+			HtmlWebpackPluginImport = undefined;
+		}
+		if (HtmlWebpackPluginImport && HtmlWebpackPluginImport.default) {
+			const HtmlWebpackPlugin = HtmlWebpackPluginImport.default;
+			const hooks = HtmlWebpackPlugin.getCompilationHooks(compilation);
+
+			if (this.options.injectSpritesInTemplates) {
+				hooks.afterTemplateExecution.tapAsync({name: 'SvgChunkWebpackPlugin'}, (data, callback) => {
+					const insertPosition = typeof this.options.injectSpritesInTemplates === 'object' ? this.options.injectSpritesInTemplates.position || 'before' : 'before';
+					const entries = typeof this.options.injectSpritesInTemplates === 'object' ? this.options.injectSpritesInTemplates.entries || [] : [];
+					if( !entries.length || entries.indexOf(data.outputName) > -1) {
+						const sprite = sprites.find((s)=>data.outputName === `${s.entryName}.html`);
+						if( sprite) {
+							if (insertPosition === 'before') {
+								data.html = data.html.replace(/<body([^>]*)>/i, `<body$1>${sprite.sprite}`);
+							} else {
+								const res= sprite.sprite.match(/^<svg([^>]*)?>/i);
+								const attributes:Record<string,string> = {};
+								if( res && res[1]) {
+									const re = /(?<property>[\w|-]+)=["']?(?<value>(?:.(?!["']?\s+(?:\S+)=|\s*\/?[>"']))+.)["']?/gi;
+									let attr
+									while(  attr = re.exec(res[1])) {
+										if (attr && attr.groups) {
+											attributes[attr.groups.property] = attr.groups.value;
+										}
+									}
+								}
+								data.bodyTags.unshift(
+									{
+										tagName: 'svg',
+										voidTag: false,
+										meta: {
+											plugin: 'svg-chunk-webpack-plugin'
+										},
+										innerHTML: sprite.sprite.replace(/<svg[^>]*?>([^$]+?)<\/svg>/mgi, '$1'),
+										attributes: attributes
+									}
+								);
```

Comment:
This part is complex, so why can't it be as simple as the `before` case?

Reply:
> This part is complex, so why can't it be as simple as the `before` case?

because I use the HtmlWebpackPlugin API which required me to define the svg tag as one of their HtmlTagObject so it forces me to 

- extract attributes from the already generated markup to insert it the attributes object
- extract the content of the svg markup to add it as innerHtml property

for sure I may have been ugly and just do a string replace it would have been way easier :)

Reply:
Question, do we really gain by using the HtmlWebpackPlugin API? Natively with the webpack API it wouldn't be feasible/simpler (I'm not familiar with that part).

Okay but why is injection simpler in the `before` case? (one line compare to +20 lines)

Reply:
> Okay but why is injection simpler in the before case? (one line compare to +20 lines)
Simply because as it is not handled by HtmlWebpackPlugin API I did a simple and "ugly" string replace.

Just made a push that did the same for `after` case.